### PR TITLE
search jobs: implement stats resolver

### PR DESCRIPTION
**Pull request by stefanhengl**
Contrary to the original plan, the stats resolver returns the aggregate state of the repo-rev jobs, NOT the repo jobs. This is easier to implement, because we don't have to worry about aggregating first on the repo level and then the job level. Let's see how the user experience is based on these stats and improve it if needed.

Notes:
- I had to add a constructor for `searchJobsResolver`, otherwise it is too easy to forget setting a field.
- While converting the call sites to use the contstrutor I came across an (now) unused searchJobsConnectionResolver, which I deleted.

Test plan: 
- updated unit test

### CODE REVIEW

File: internal/search/exhaustive/store/exhaustive_search_jobs.go [hunk 1]
```diff
@@ -303,6 +303,48 @@ func (s *Store) DeleteExhaustiveSearchJob(ctx context.Context, id int64) (err er
 	return s.Exec(ctx, sqlf.Sprintf(deleteExhaustiveSearchJobQueryFmtStr, id))
 }
 
+const getAggregateRepoRevState = `
+SELECT rrj.state, count(rrj.id) as count FROM exhaustive_search_repo_revision_jobs rrj
+JOIN exhaustive_search_repo_jobs rj ON rrj.search_repo_job_id = rj.id
+JOIN exhaustive_search_jobs sj ON rj.search_job_id = sj.id
+WHERE sj.id = %s
+GROUP BY rrj.state
```
Comment (keegancsmith) [inline internal/search/exhaustive/store/exhaustive_search_jobs.go:311]: only concern is that until we actually start running this job the stats would be empty. It also won't communicate other failures that can occur. I think merge this PR as is, but maybe experiment in a follow up with aggregating these stats across all the tables in question?

No idea if this is good sql or if it works, but maybe something like this?

``` sql
SELECT state, count(*) as count FROM
  (
   (SELECT state FROM exhaustive_search_jobs ...)
   UNION
   (SELECT state FROM exhaustive_search_repo_jobs ...)
   UNION
   (SELECT state FROM exhaustive_search_repo_revision_jobs ...)
  )
GROUP BY state
```

File: internal/search/exhaustive/store/exhaustive_search_jobs.go [hunk 2]
```diff
@@ -303,6 +303,48 @@ func (s *Store) DeleteExhaustiveSearchJob(ctx context.Context, id int64) (err er
 	return s.Exec(ctx, sqlf.Sprintf(deleteExhaustiveSearchJobQueryFmtStr, id))
 }
 
+const getAggregateRepoRevState = `
+SELECT rrj.state, count(rrj.id) as count FROM exhaustive_search_repo_revision_jobs rrj
+JOIN exhaustive_search_repo_jobs rj ON rrj.search_repo_job_id = rj.id
+JOIN exhaustive_search_jobs sj ON rj.search_job_id = sj.id
```
Comment (keegancsmith) [inline internal/search/exhaustive/store/exhaustive_search_jobs.go:309]: you don't need to join on this table, you can just directly do it on rj.search_job_id?

File: internal/search/exhaustive/store/exhaustive_search_jobs.go [hunk 3]
```diff
@@ -303,6 +303,48 @@ func (s *Store) DeleteExhaustiveSearchJob(ctx context.Context, id int64) (err er
...
+WHERE sj.id = %s
+GROUP BY rrj.state
+`
+
+func (s *Store) GetAggregateRepoRevState(ctx context.Context, id int64) (_ map[string]int, err error) {
```
Comment (keegancsmith) [inline internal/search/exhaustive/store/exhaustive_search_jobs.go:314]: It feels like this type should involve the enum in the return and not generic string?

**Review by keegancsmith — APPROVED** [review]
LGTM

side note: we should rename these tables before we have customer data in them

File: internal/search/exhaustive/store/exhaustive_search_jobs.go (hunk 3 above)
Comment (stefanhengl) [reply internal/search/exhaustive/store/exhaustive_search_jobs.go:314]: Do you mean the `types.JobState` type like map[types.JobState]int?

Note: The worker framework doesn't seem to have an enum for the state so we don't really know what is coming back from the db.

**Pull request merged**